lib/api.js

```diff
@@ -18,6 +18,7 @@ function PlexAPI(options, deprecatedPort) {
 
     this.hostname = hostname;
     this.port = deprecatedPort || opts.port || PLEX_SERVER_PORT;
+    this.https = opts.https || false;
```

Comment:
```js
this.https = opts.https;
```
Could that possibly fix the (rare) issue you're describing?

> There is a teeny tiny corner case issue which I think is acceptable for the sake of backwards compatibility: if you supply both port: 443 and https: false, it will still use https due to the port number.

Cause then line #197 could be
```js
return this.https === true || (this.port === 443 && this.https === undefined) ? 'https://' : 'http://';
```